File: 1300/2118C/main.java
import java.util.*;
public class main
{
	public static void main(String[] args) 
	{
		Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();
 
		while(t-- > 0)
		{
			int n = sc.nextInt();
			long k = sc.nextLong();
 
			long arr[] = new long[n];
			for(int i=0;i<n;i++)
				arr[i] = sc.nextInt();
 
			long sum = 0;
			long ans = 0;
 
			for(int i=0;i<63;i++) 
			{
				PriorityQueue<Long> pq = new PriorityQueue<>();
				long num = (long)1<<i;
				
				for(int j=0;j<n;j++) 
				{
					long a = arr[j];
					if((num&a) > 0) 
						ans++;
					else
						pq.add(num);
				}
 
				while(!pq.isEmpty() && k >= pq.peek())
				{
					long a = pq.poll();
					ans += 1;
					k -= a;
				}
			}
 
			System.out.println(ans);
		}
	}
	
}

File: 1300/2118C/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n;
        long long k;
        cin >> n >> k;

        vector<long long> arr(n);
        for (int i = 0; i < n; i++)
            cin >> arr[i];

        long long ans = 0;

        for (int i = 0; i < 63; i++) 
        {
            priority_queue<long long, vector<long long>, greater<long long>> pq;
            long long num = 1LL << i;

            for (int j = 0; j < n; j++) 
            {
                long long a = arr[j];
                if ((num & a) > 0)
                    ans++;
                else
                    pq.push(num);
            }

            while (!pq.empty() && k >= pq.top()) 
            {
                long long a = pq.top();
                pq.pop();

                ans++;
                k -= a;
            }
        }

        cout << ans << "\n";
    }

    return 0;
}
